lgsm/functions/core_dl.sh

```diff
@@ -58,7 +58,7 @@ fn_dl_extract(){
 	mime=$(file -b --mime-type "${filedir}/${filename}")
 
 	if [ "${mime}" == "application/gzip" ]||[ "${mime}" == "application/x-gzip" ]; then
-		tarcmd=$(tar -zxf "${filedir}/${filename}" -C "${extractdir}")
+		tarcmd=$(tar -zxf "${filedir}/${filename}" -C "${extractdir}" --strip-components=1)
```

Comment:
--strip-components=NUMBER
       strip NUMBER leading components from file names on extraction

Basically it removes the leading, empty directory from the tar file. In which MTA has.

It depends on the other extractions. However I'm not sure how else to handle this while still using your extract function since I can't pass arguments to it. If it's acceptable to not use your extract function then I can do it differently.